### ISSUE TITLE
make: arch: Command not found

### ISSUE THREAD

**Issue by sebadoom** [google/fscrypt#92]
During build I am now seeing the following error:

```
make: arch: Command not found
go build --ldflags '-s -w -X "main.version=v0.2.3-2-gbd2ca31" -X "main.buildTime=Wed Feb 14 13:21:43 -03 2018" -extldflags "-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now"' -o bin/fscrypt ./cmd/fscrypt
go build -buildmode=c-shared --ldflags '-s -w -X "main.version=v0.2.3-2-gbd2ca31" -X "main.buildTime=Wed Feb 14 13:21:43 -03 2018" -extldflags "-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now"' -o bin/pam_fscrypt.so ./pam_fscrypt
rm -f bin/pam_fscrypt.h
```

The build succeeds nonetheless. Is this expected? It might be related to commit 3326520d48c6fd4b65c5b9d19f7eb8a005f6b943

**Comment by josephlr** [comment]
[`arch(1)`](http://man7.org/linux/man-pages/man1/arch.1.html) is a command line tool that just prints out the architecture you're currently running (this is so the right version of the [Protobuf Compiler](https://github.com/google/protobuf) can be used in the Makefile). The command is part of [coreutils](url) and was [introduced there in 2007](https://github.com/google/protobuf) for version 6.10. You can check your version of coreutils by running `ls --version`.

Even though it's been around for awhile, `arch` does the exact same thing as `uname -m`, so I'll just use that instead, and it should fix your problem.

Also, out of curiosity, what system are you running?

**Comment by sebadoom** [comment]
ArchLinux, up-to-date. I don't have the `arch` command and I do have coreutils, of course:

```
seba@i7-4790K-LIN [04:01:16 PM] [~] 
-> % arch
zsh: command not found: arch
seba@i7-4790K-LIN [04:01:27 PM] [~] 
-> % pacman -Qi coreutils 
Name            : coreutils
Version         : 8.29-1
Description     : The basic file, shell and text manipulation utilities of the GNU operating system
Architecture    : x86_64
URL             : https://www.gnu.org/software/coreutils/
Licenses        : GPL3
Groups          : base
Provides        : None
Depends On      : glibc  acl  attr  gmp  libcap  openssl
Optional Deps   : None
Required By     : ca-certificates-utils  intellij-idea-community-edition  linux  mkinitcpio  netctl  util-linux
Optional For    : usbutils
Conflicts With  : None
Replaces        : None
Installed Size  : 13.79 MiB
Packager        : Sébastien Luttringer <seblu@seblu.net>
Build Date      : Fri 29 Dec 2017 09:08:09 AM -03
Install Date    : Mon 22 Jan 2018 03:32:13 PM -03
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : Signature
```

**Comment by sebadoom** [comment]
Ah, here's what's going on: `arch` is not installed by default according to coreutils' docs: https://www.gnu.org/software/coreutils/manual/html_node/arch-invocation.html

```
arch is not installed by default, so portable scripts should not rely on its existence.
```

It needs to be explicitly enabled using `configure`. As you said, `uname -m` might be more portable.

**Comment by josephlr** [comment]
Ahhhhhhh... That makes sense. I guess I'll send in a patch to the man pages then. This seems like the type of thing that should be mentioned.

**Comment by josephlr** [comment]
Should be fixed in 7442ff1144c91ba9810b66fa843baf8c1953107f
